apps/frontend/src/modules/notification/notification-api.ts

```diff
@@ -0,0 +1,36 @@
+import { APIPath, ContentType } from "~/libs/enums/enums.js";
+import { BaseHTTPApi } from "~/libs/modules/api/api.js";
+import { type HTTP } from "~/libs/modules/http/http.js";
+import { type Storage } from "~/libs/modules/storage/storage.js";
+import { type APIResponse } from "~/libs/types/types.js";
+
+import { NotificationApiPath } from "./libs/enums/enums.js";
+import { type NotificationGetAllItemResponseDto } from "./libs/types/types.js";
+
+type Constructor = {
+	baseUrl: string;
+	http: HTTP;
+	storage: Storage;
+};
+
+class NotificationApi extends BaseHTTPApi {
+	public constructor({ baseUrl, http, storage }: Constructor) {
+		super({ baseUrl, http, path: APIPath.NOTIFICATIONS, storage });
+	}
+
+	public async getNotifications(): Promise<
```

Comment:
By quality criteria, we need to avoid entity name repetition for methods if it's clear from the className (NotificationsApi)
```suggestion
	public async getAll(): Promise<
```